### PR TITLE
Update readme to match new template

### PR DESCRIPTION
**Pull request by laura-rodriguez**

### CODE REVIEW

**Comment by nbarbettini** [comment]
I marked a few "TODOs" and also rebased + cleaned up the PR so it will be a clean merge to master.

File: README.md [hunk 1]
```diff
@@ -1,166 +1,201 @@
...
+});
+```
+
+### Add a User to a Group
+``` csharp
+// Find the desired user
+var user = await client.Users.FirstOrDefault(x => x.Profile.Email == "darth.vader@imperial-senate.gov");
 
 // find the desired group
 var group = await _oktaClient.Groups.FirstOrDefault(x => x.Profile.Name == "Stormtroopers");
```
Comment (nbarbettini) [inline README.md]: Change to `await client`

File: README.md [hunk 2]
```diff
@@ -172,58 +207,189 @@ if (group != null && user != null)
...
-await _oktaClient.PostAsync(
-    new Okta.Sdk.Internal.HttpRequest
+// Verify sms factor
+var response = await client.UserFactors.VerifyFactorAsync(verifyFactorRequest, user.Id, smsFactor.Id);
+```
+
+### List all Applications
+``` csharp
+// List all applications
+var appList = await client.Applications.ListApplications().ToArray();
```
Comment (nbarbettini) [inline README.md]: `await client.Applications.ToArray()` should work here right? I prefer the simpler syntax.

File: README.md [hunk 3]
```diff
@@ -1,166 +1,201 @@
...
-await _oktaClient.Groups.CreateGroupAsync
-(
-    new CreateGroupOptions()
-    {
-        Name = "Stormtroopers",
-        Description = "Some description here..."
-    }
-);
+// find the desired user
+var user = await _oktaClient.Users.FirstOrDefault(x => x.Profile.Email == "laura.rodriguez@okta.com");
```
[the remark this answers is not in the document]
Comment (laura-rodriguez) [reply README.md]: Have a hole in my mind, when/why did I provide my email?

**Comment by nbarbettini** [comment]
It's definitely available for Users and Groups, since those interfaces
implement IAsymcEnumerable. Maybe we missed doing that for Applications?

File: README.md (hunk 2 above)
Comment (laura-rodriguez) [reply README.md]: `ApplicationsClient` implements ` IAsyncEnumerable<IApplication>` but I forgot to add this in the `IApplicationsClient`.  I will push this change before publishing on Nuget, will tell Rob to review.

**Pull request merged**